lib/fluent/plugin/out_http.rb

```diff
@@ -64,7 +64,7 @@ class RetryableResponse < StandardError; end
     desc 'Raise UnrecoverableError when the response is non success, 4xx/5xx'
     config_param :error_response_as_unrecoverable, :bool, default: true
     desc 'The list of retryable response code'
-    config_param :retryable_response_codes, :array, value_type: :integer, default: [503]
+    config_param :retryable_response_codes, :array, value_type: :integer, default: [500, 502, 503, 504]
```

Comment:
@ganmacs thanks for feedback. Did you read the commit message?

> 500 is an internal server error that occurred when an error happens in a server. It'll be helpless to retry.

It's not helpless in the majority of the cases. I argue that in practice, in the majority of the cases (not always),  a 500 HTTP response represents a **transient** error. This is based on experience writing such systems, operating such systems, and debugging them. I don't like to put this into numbers, but this is not even 50/50 but more like 10/90 (90 % of the cases and more the underlying problem is transient).

In fact, I am working right now with a sane system for which I specifically added the 500 to the list of `retryable_response_codes` for the out_http plugin. And because I am convinced that this makes sense by default I am merely offering to contribute this as a patch :).

Of course, rarely you will find systems where a 500 error represents a permanent error. As of a bug in code, for example.

So: I am proposing this change with the idea to have settings that help in the majority of environments :-).

Most 4xx responses should not be retried, sure, because their whole point is to say: something is bad with your request.

The point of most 5xx responses is to say that nothing is wrong with the request, but that the problem is in the other end, and that this problem is of temporary nature.

About any web application framework (be it for Ruby, Java, Python, Go, ...) emits a 500 upon uncaught errors, and transient errors are often consciously uncaught, such as TCP connect() timeouts, or TCP recv() errors or timeouts.

Reply:
In any case, no hard feelings. I take "no" for an answer.

Reply:
Oh sorry, I didn't read it and completely agreed with your surprise.

> We were surprised to find that the HTTP output
plugin by default considers a 500 HTTP response
to be unrecoverable.

> Of course, rarely you will find systems where a 500 error represents a permanent error. As of a bug in code, for example.

This is my point. we can't decide if the error is retriable or not. and then, we should not add the 5xx in default retriable error event even if almost 500 error can be retriable in practice. 

IMO, I think it makes sense for me to remove 503 from retriable errors.
WDYT? @repeatedly 

Thank you for your suggestion :)

Reply:
> IMO, I think it makes sense for me to remove 503 from retriable errors.

It breaks existing environment. We need to wait fluentd v2 for such breaking changes.

Reply:
I know that. My point here is that you are for or against the idea which removes 503 from default retriable error.

Reply:
@repeatedly WDYT?

Reply:
503 is used for temporal overloaded situation, so `out_http` has 503 as default value.
If this is not fit for fluentd case, removing 503 is reasonable.